tests/unit/test_main.py

```diff
@@ -81,41 +81,37 @@ def load_plugins_mock(mocker, monkeypatch):
     return load_plugins_mock
 
 
-args = 'error_type'
-params = [
-    'list', 'obj'
-]
-
-
-@pytest.mark.parametrize(args, params)
-def test_log_or_exit_on_exceptions_no_debug(error_type, plugin_exc_mock,
-                                            mocker, monkeypatch):
+@pytest.mark.parametrize('error_type', ['list', 'obj'])
+def test_log_or_exit_on_exceptions_no_debug(
+        error_type, plugin_exc_mock, caplog, capsys):
     """Raise SystemExit if debug flag is off."""
-    logging_mock = mocker.MagicMock(main.logging, autospec=True)
-    monkeypatch.setattr(main, 'logging', logging_mock)
-
-    error = ('bad.plugin', plugin_exc_mock)
+    error = plugin_exc_mock
     if error_type == 'list':
-        error = [('bad.plugin', error)]
+        error = [error]
     with pytest.raises(SystemExit) as e:
         main._log_or_exit_on_exceptions('base msg', error, debug=False)
-
     e.match('1')
-    logging_mock.error.assert_called_once()
-    logging_mock.warn.assert_not_called()
 
+    assert 1 == len(caplog.records)
+    for record in caplog.records:
+        assert 'ERROR' == record.levelname
 
-def test_log_or_exit_on_exceptions_debug(plugin_exc_mock, mocker, monkeypatch):
-    """Do not exit out if debug flag is on."""
-    logging_mock = mocker.MagicMock(main.logging, autospec=True)
-    monkeypatch.setattr(main, 'logging', logging_mock)
+    out, err = capsys.readouterr()
+    assert '--- Logging error ---' not in err
 
-    errors = [('bad.plugin', plugin_exc_mock)]
 
-    main._log_or_exit_on_exceptions('base_msg', errors, debug=True)
+def test_log_or_exit_on_exceptions_debug(
```

Comment:
Should this also be parametrized?

Reply:
On error type?  I don't think there is value as it doesn't increase code coverage.

The main change I made here was to replace the logging mock with capsys since the discovery of the bug has emphasized the danger in overusing mocks. I also added the assert of no logging error, which isn't strictly needed because it's tested above, and I can remove it if causes confusion.  I added it because I actually changed this test first, since it was less complicated, but it doesn't really need to be in both tests.

Reply:
Well, parametrizing on error type does capture more of the input space, but it's not super necessary, just a thought.  I do think if we're limiting this test to things that aren't captured by the other one, removing the second assert is probably a good idea.

BTW ISTR that there's a way to do 'with pytest.raises(None)' or something that you could use to also explicitly test that it doesn't raise.

Reply:
I can remove the assert. I think `pytest.raises(None)` adds complexity where it is not needed.